server/routes/summary_routes.ts

```diff
@@ -17,6 +17,35 @@ const LOG_PATTERN_SUMMARY_AGENT_CONFIG_ID = 'os_summary_with_log_pattern';
 const OS_INSIGHT_AGENT_CONFIG_ID = 'os_insight';
 const DATA2SUMMARY_AGENT_CONFIG_ID = 'os_data2summary';
 
+export function postProcessing(output: string) {
+  const summarizeStartTag = '<summarization>';
+  const summarizeEndTag = '</summarization>';
+  const startTag = '<final insights>';
+  const endTag = '</final insights>';
+
+  if (
+    output.includes(summarizeStartTag) &&
+    output.includes(summarizeEndTag) &&
+    output.includes(startTag) &&
+    output.includes(endTag)
+  ) {
+    // Extract <summarization>
+    const summarizationStart = output.indexOf(summarizeStartTag) + summarizeStartTag.length;
+    const summarizationEnd = output.indexOf(summarizeEndTag);
+    const summarization = output.substring(summarizationStart, summarizationEnd).trim();
+
+    // Extract <final insights>
+    const insightsStart = output.indexOf(startTag) + startTag.length;
+    const insightsEnd = output.indexOf(endTag);
+    const finalInsights = output.substring(insightsStart, insightsEnd).trim();
+
+    // Combine <summarization> and <final insights>
+    const processedOutput = `${summarization}\n${finalInsights}`;
+    return processedOutput;
+  }
+  return output;
+}
```

Comment:
nit: the logic looks good to me, we can use regex to make it more clear

do we need to consider the case for tags are upper case?

```suggestion
  const pattern = /<summarization>(.*?)<\/summarization>.*?<final insights>(.*?)<\/final insights>/s;

  const match = output.match(pattern);
  if (match) {
    // Extract and combine summarization and insights
    const [, summarization, finalInsights] = match;
    const processedOutput = `${summarization.trim()}\n${finalInsights.trim()}`;
    return processedOutput;
  }
  return output;
```

Reply:
yes we definitly can